File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20191103_135723681/ネームバトラー【マスター編】(学習3)/3.３対３のパーティー対戦にする/Fighter.java
package name_battlerM3;

// プレイヤー：戦士
public class Fighter extends Player {

	// =======================
	// フィールド変数
	// =======================


	// =======================
	// コンストラクタ
	// =======================
	public Fighter(String name)
	{
		super(name);
	}

	// =======================
	// Getter / Setter
	// =======================


	// =======================
	// protected メソッド
	// =======================
	/**
	 * 名前(name)からキャラクターに必要なパラメータを生成する
	 */
	@Override
	protected void MakeCharacter()
	{
		// 戦士のパラメータを名前から生成する
		this.hp = GetNumber(0, 200) + 100;
		this.mp = 0;
		this.str = GetNumber(1, 30) + 70;
		this.def = GetNumber(2, 30) + 70;
		this.luck = GetNumber(3, 99) + 1;
		this.agi = GetNumber(4, 49) + 1;
	}

	/**
	 * 対象プレイヤーに攻撃を行う
	 * @param defender : 対象プレイヤー
	 */
	@Override
	public void Attack(Player defender)
	{
        // 与えるダメージを求める
        System.out.println(GetName() + "の攻撃！");
        int damage = CalcDamage(defender);

        // 求めたダメージを対象プレイヤーに与える
        System.out.println(defender.GetName() + "に" + damage + "のダメージ！");
        defender.Damage(damage);

        // 倒れた判定
        if (defender.GetHP() <= 0) {
            System.out.println(defender.GetName() + "は力尽きた...");
        }
	}


	// =======================
	// private メソッド
	// =======================


	// =======================
	// public メソッド
	// =======================
}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Fighter.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Fighter : Player
{

	// =======================
	// フィールド変数
	// =======================
	// =======================
	// コンストラクタ
	// =======================
	public Fighter(string name) : base(name)
	{
		this.name = name;
		MakeCharacter();
	}

	// =======================
	// Getter / Setter
	// =======================

	// =======================
	// protected メソッド
	// =======================

	/// <summary>
	/// <inheritdoc/>
	/// </summary>
	/// 
	protected override void MakeCharacter()
	{
		// 戦士のパラメータを名前から生成する
		this.defaultHP = (base.GetNumber(0, 30) + 10) * 10;
		this.defaultMP = base.GetNumber(1, 100) * 0;
		this.hp = base.defaultHP;
		this.mp = base.defaultMP;

		this.str = base.GetNumber(2, 100) + 30;
		this.def = base.GetNumber(3, 100) + 30;
		this.luck = base.GetNumber(4, 100) + 1;
		this.agi = base.GetNumber(5, 50) + 1;

		this.paralyze = false;
		this.poison = false;
		this.active = true;


	}


	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================

	/// <summary>
	/// <inheritdoc/>
	/// </summary>
	/// <param name="activePlayer"><inheritdoc/></param>
	/// <param name="passiveMembers"><inheritdoc/></param>
	public override void Attack(Player activePlayer, List<Player> passiveMembers)
	{

		//攻撃対象の決定
		Player passivePlayer = passiveMembers[0];

		if (passiveMembers.Count > 1)
		{
			passivePlayer = passiveMembers[UnityEngine.Random.Range(0, passiveMembers.Count - 1)];

		}

		// 与えるダメージを求める
		LogText.AddLog(string.Format("{0} の攻撃！", activePlayer.GetName()));
		int damage = activePlayer.CalcDamage(passivePlayer);

		// 求めたダメージを対象プレイヤーに与える
		LogText.AddLog(string.Format("{0}に {1} のダメージ！", passivePlayer.GetName(), damage));
		passivePlayer.Damage(damage);

		passivePlayer.Down();
	}
}